packages/template-retail-react-app/app/pages/checkout/partials/contact-info.jsx

```diff
@@ -69,6 +70,12 @@ const ContactInfo = () => {
                 })
             } else {
                 await login.mutateAsync({username: data.email, password: data.password})
+
+                // Because we lazy load the basket there is no guarantee that a basket exists for the newly registered
+                // user, for this reason we must transfer the ownership of the previous basket to the logged in user.
+                await transferBasket.mutateAsync({
+                    parameters: {overrideExisting: true}
+                })
```

Comment:
is there anywhere else that we should transfer basket upon login? does it make sense to always transfer basket upon login? (not only in checkout contact info)

Reply:
What I think you are eluding to is the `mergeBasket` action. We currently already have some logic in the login form action for this. Logging in during checkout is a little different as you don't want unexpected items in your cart when you are essentially already in the checkout flow. 

For example, we don't want a guest user to login at the time of checkout and see that the item added in their basket during the guest shopping was added to an old basket for the registered account.

Slight differences here, but they are important.